File: Data Structures & Algorithms/kth-distinct-string-in-an-array/submission-3.py
class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        map = Counter(arr)
        
        for st in arr:
            if(map[st] == 1):
                k -= 1
                if k == 0:
                    return st
        return ""
            

File: Data Structures & Algorithms/find-k-closest-elements/submission-3.cpp
class Solution {
public:
    vector<int> findClosestElements(vector<int>& arr, int k, int x) {
        int st = 0;
        for(int i = 1 ; i <= arr.size()-k; i++) {
            int l = abs(arr[st]-x);
            int r = abs(arr[i+k-1]-x);
           
            if(r < l) {
                st = i;
            }
        }
        vector<int> ans(arr.begin()+st, arr.begin()+st+k);
        return ans;
    }
};